sql/core/src/test/resources/log4j.properties

```diff
@@ -22,7 +22,7 @@ log4j.rootLogger=INFO, CA, FA
 log4j.appender.CA=org.apache.log4j.ConsoleAppender
 log4j.appender.CA.layout=org.apache.log4j.PatternLayout
 log4j.appender.CA.layout.ConversionPattern=%d{HH:mm:ss.SSS} %p %c: %m%n
-log4j.appender.CA.Threshold = WARN
+log4j.appender.CA.Threshold = FATAL
```

Comment:
oversized log for GA console output cause truncation fo useful test errors, andthe unit-tests.log is enough

Reply:
If so, Is it better to backport this change into the previous branches? cc: @HyukjinKwon @dongjoon-hyun

Reply:
yeah, I guess so. we can mute console output for most modules. we call `o.s.Assertions.intercept` frequently which produces a lot of unnecessary error logs. 

When we mute them, the error stacktraces for failed tests can still be kept.

I can make a separate PR to fix it if it makes sense to the CCers.

Reply:
But the current threshold (WARN) can help catch a bug like the following early: https://github.com/apache/spark/pull/31273#issuecomment-785854571

Reply:
> But the current threshold (WARN) can help catch a bug like the following early: [#31273 (comment)](https://github.com/apache/spark/pull/31273#issuecomment-785854571)

These warning messages still can be found in the `unit-tests.log`s. I don't see much difference as a warning message can still be simply ignored.

But when we encounter test failures but got omitted by the CI, it is hard for us to locate  to them.

Reply:
I think it's okay as long as `unit-tests.log` contains.

Reply:
Thanks, I'll send a PR then